lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -77,45 +106,155 @@ def __init__(
-            term = {"authorization.owner": user}
-        elif json.get("access") == Dataset.PRIVATE_ACCESS:
-            raise UnsupportedAccessMode(user, Dataset.PRIVATE_ACCESS)
+            user_term = {"term": {"authorization.owner": user}}
+
+        # IF we're looking at a user we're not authorized to see (either the
+        # call is unauthenticated, or authenticated as a non-ADMIN user trying
+        # to reference a different user), then we're only allowed to see PUBLIC
+        # data; private data requests aren't allowed, and we expect these cases
+        # to fail with appropriate permission errors (401 or 403) before we
+        # reach this code.
+        #
+        # ELSE IF we're looking for a specific access category, add the query
+        # term.
+        #
+        # ELSE IF this is a {} query from a non-ADMIN user, add a "disjunction"
+        # (OR) covering all the user's own data plus all public data.
+        #
+        # ELSE this must be an admin user looking for data owned by a specific
+        # user (there's no access parameter, and possibly no user parameter),
+        # and we'll pass through the query either with the specified user
+        # constraint or with no constraint at all.
+        if not authorized_user or (user and user != authorized_id and not is_admin):
+            if access == Dataset.PRIVATE_ACCESS:
+                requestor = (
+                    f"Requestor {authorized_user.username!r}"
+                    if authorized_user
+                    else "Unauthorized requestor"
+                )
+                raise AssemblyError(
+                    f"Internal error: {requestor} can't query private data"
+                )
+            access_term = {"term": {"authorization.access": Dataset.PUBLIC_ACCESS}}
+            self.logger.debug("QUERY: not self public: {}", filter)
+        elif access:
+            access_term = {"term": {"authorization.access": access}}
+            if not user and access == Dataset.PRIVATE_ACCESS and not is_admin:
+                user_term = {"term": {"authorization.owner": authorized_id}}
+            self.logger.debug("QUERY: access: {}", filter)
+        elif authorized_user and not user and not is_admin:
+            filter.append(
+                {
+                    "dis_max": {
+                        "queries": [
+                            {"term": {"authorization.owner": authorized_id}},
+                            {"term": {"authorization.access": Dataset.PUBLIC_ACCESS}},
+                        ]
+                    }
+                }
+            )
+            self.logger.debug("QUERY: {{}} default: {}", filter)
+            return {"bool": {"filter": filter}}
         else:
-            term = {"authorization.access": Dataset.PUBLIC_ACCESS}
-        return term
+            # Either "user" was specified and is already added to the filter,
+            # or client is ADMIN and no access terms are required.
+            pass
```

Comment:
The boolean logic above seems "okay" but it is hard to prove we have all the cases.  Anyway we can record how many times we get to `pass` to help understand the behavior of the queries?

Reply:
Adding a `debug` log entry would be great.

I have not reviewed the tests yet, but having a `debug` log will help us catch behavioral drift over time.